instance/instance.go

```diff
@@ -143,119 +145,136 @@ func (s *Store) Update(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, err.Error(), http.StatusForbidden)
 			return
 		}
+
+		// TODO Update dataset.next state to associated and add collection id
 	case publishedState:
 		if err = validateInstanceUpdate(associatedState, currentInstance, instance); err != nil {
 			log.Error(err, log.Data{"instance_id": id, "current_state": currentInstance.State})
 			http.Error(w, err.Error(), http.StatusForbidden)
 			return
 		}
-	}
 
-	if err = s.UpdateInstance(id, instance); err != nil {
-		log.Error(err, nil)
-		handleErrorType(err, w)
-		return
+		// TODO Update both edition and dataset states to published
 	}
 
 	if instance.State == editionConfirmedState {
 		var editionDoc *models.Edition
 
 		datasetID := currentInstance.Links.Dataset.ID
-		edition := currentInstance.Edition
-		if instance.Edition != "" {
-			edition = instance.Edition
+		instance.Links.Job = currentInstance.Links.Job
+
+		// If instance has no edition, get the current edition
+		if instance.Edition == "" {
+			instance.Edition = currentInstance.Edition
 		}
 
-		// Set the instance edition to the latest version
-		instance.Edition = edition
+		edition := instance.Edition
 
-		// create unique id for edition
-		editionID := (uuid.NewV4()).String()
+		// Only create edition if it doesn't already exist
+		editionDoc, err = s.GetEdition(datasetID, edition, "")
+		if err != nil {
+			if err != errs.EditionNotFound {
+				log.Error(err, nil)
+				handleErrorType(err, w)
+			}
 
-		editionDoc = &models.Edition{
-			ID:      editionID,
-			Edition: edition,
-			Links: models.EditionLinks{
-				Dataset: models.LinkObject{
-					ID:   datasetID,
-					HRef: fmt.Sprintf("%s/datasets/%s", s.Host, datasetID),
+			// create unique id for edition
+			editionID := (uuid.NewV4()).String()
+
+			editionDoc = &models.Edition{
+				ID:      editionID,
+				Edition: edition,
+				Links: &models.EditionLinks{
+					Dataset: &models.LinkObject{
+						ID:   datasetID,
+						HRef: fmt.Sprintf("%s/datasets/%s", s.Host, datasetID),
```

Comment:
Maybe we need a story to go through all our services and sanitise our urls, whether that be on input (like here) or output (when services make http requests to our API's).

I don't want to start this here, as there are many cases in this repository where this would need to be updated.